legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabEntityApi.java

```diff
@@ -923,5 +939,18 @@ synchronized Entity getEntity()
             }
             return this.entity;
         }
+
+        synchronized Entity getInvalidEntity()
```

Comment:
As with EntityAccessContext, this cannot return Entity. If the entity is indeed invalid, you should cache that fact and check it both in this method and in getEntity.

Also, this needs to check for any kind of invalidity, not just a path mismatch. If an entity cannot be deserialized for any reason, it is invalid.